File: 219.存在重复元素-ii.cpp
/*
 * @lc app=leetcode.cn id=219 lang=cpp
 *
 * [219] 存在重复元素 II
 */

// @lc code=start
class Solution {
  public:
    bool containsNearbyDuplicate(vector<int> &nums, int k) {
        map<int, int> m;
        for (int i = 0; i < nums.size(); i++) {
            if (m.count(nums[i])) {
                if (i - m[nums[i]] > k) {
                    m[nums[i]] = max(m[nums[i]], i);
                }
                else {
                    return true;
                }
            }
            else {
                m[nums[i]] = i;
            }
        }
        return false;
    }
};
// @lc code=end

File: 219.存在重复元素-ii.go
/*
 * @lc app=leetcode.cn id=219 lang=golang
 *
 * [219] 存在重复元素 II
 */

// @lc code=start

func containsNearbyDuplicate(nums []int, k int) bool {
	m := map[int]int {}
	for i, n := range nums {
		if m[n] !=0 {
			if math.Abs(float64(i+1-m[n])) <= float64(k) {
				return true
			}else {
				m[n] = i + 1
			}
		} else {
			m[n] = i + 1
		}
	}
	return false
}

// @lc code=end

